api/machine_types.go

```diff
@@ -601,7 +601,7 @@ type MachineConfig struct {
 	Schedule string `json:"schedule,omitempty"`
 	// Optional boolean telling the Machine to destroy itself once it’s complete (default false)
 	AutoDestroy bool             `json:"auto_destroy,omitempty"`
-	Restart     MachineRestart   `json:"restart,omitempty"`
+	Restart     *MachineRestart  `json:"restart,omitempty"`
```

Comment:
just to raise a warning, this change will need some changes in flaps side as it imports flyctl/api module

Reply:
How do we usually do that @dangra? I'd have to merge this first right?

Reply:
Yes, that is one way of doing it but only if you plan to do it right after merging the PR. Otherwise pull the updated flyctl/api dependency from this branch, prepare the changes and merge once it is all ready.